snapcraft/internal/pluginhandler/__init__.py

```diff
@@ -1087,10 +1086,37 @@ def check_for_collisions(parts):
                     conflict_files=conflict_files)
 
         # And add our files to the list
-        parts_files[part.name] = {'files': part_files,
+        parts_files[part.name] = {'files': part_contents,
                                   'installdir': part.installdir}
 
 
+def _paths_collide(path1: str, path2: str) -> bool:
+    if not (os.path.lexists(path1) and os.path.lexists(path2)):
+        return False
+
+    path1_is_dir = os.path.isdir(path1)
+    path2_is_dir = os.path.isdir(path2)
+    path1_is_link = os.path.islink(path1)
+    path2_is_link = os.path.islink(path2)
+    collide = False
```

Comment:
instead of this, why not add an 
```python
else:
    collide = False
```
to the end of your conditionals (it would feel more complete).

Reply:
Done. I actually got rid of `collide` all together.